### ISSUE TITLE
Issues with gcc-8

### ISSUE THREAD

**Issue by leezu** [intel/dptf#23]
cmake works fine
```

% cmake ..                                                                             ~/software/dptf/DPTF/Linux/build master
-- The C compiler identification is GNU 8.2.0
-- The CXX compiler identification is GNU 8.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CXX Flags (initial) :
Building for Linux...
Building 64-bit release...
GCC version : 8.2.0
CXX Flags (final) :  -std=c++11 -Wall -Werror -Wno-unused-value -fPIC -Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector-strong -m64
CXX Debug Flags : -g -O0
CXX Release Flags : -Os
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/leonard/software/dptf/DPTF/Linux/build
```

But actual compilation fails

```
% make                                                                                 ~/software/dptf/DPTF/Linux/build master
[  1%] Building CXX object SharedLib/CMakeFiles/DptfSharedLib.dir/home/leonard/software/dptf/DPTF/Sources/SharedLib/BasicTypesLib/AppVersion.cpp.o
<command-line>: error: "_FORTIFY_SOURCE" redefined [-Werror]
<built-in>: note: this is the location of the previous definition
cc1plus: all warnings being treated as errors
make[2]: *** [SharedLib/CMakeFiles/DptfSharedLib.dir/build.make:63: SharedLib/CMakeFiles/DptfSharedLib.dir/home/leonard/software/dptf/DPTF/Sources/SharedLib/BasicTypesLib/AppVersion.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:112: SharedLib/CMakeFiles/DptfSharedLib.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

```

**Comment by leezu** [comment]
Removing `-D_FORTIFY_SOURCE=2` from the CMakeLists.txt helps. 

Compilation then gets stuck at:
```

% make                                                                        1m 18s ~/software/dptf/DPTF/Linux/build master +
[ 19%] Built target DptfSharedLib
[ 21%] Built target DptfBasicTypesLib
[ 23%] Built target DptfEsifTypesLib
[ 27%] Built target DptfDptfTypesLib
[ 28%] Built target DptfDptfObjectsLib
[ 34%] Built target DptfParticipantControlsLib
[ 36%] Built target DptfParticipantLib
[ 37%] Built target DptfEventsLib
[ 37%] Built target DptfMessageLoggingLib
[ 39%] Built target DptfXmlLib
[ 39%] Building CXX object Policies/PolicyLib/CMakeFiles/DptfPolicyLib.dir/home/leonard/software/dptf/DPTF/Sources/Policies/PolicyLib/ActiveTripPointsCachedProperty.cpp.o
/home/leonard/software/dptf/DPTF/Sources/Policies/PolicyLib/ActiveTripPointsCachedProperty.cpp: In member function ‘virtual Bool ActiveTripPointsCachedProperty::supportsProperty()’:
/home/leonard/software/dptf/DPTF/Sources/Policies/PolicyLib/ActiveTripPointsCachedProperty.cpp:76:9: error: catching polymorphic type ‘class dptf_exception’ by value [-Werror=catch-value=]
  catch (dptf_exception)
         ^~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [Policies/PolicyLib/CMakeFiles/DptfPolicyLib.dir/build.make:135: Policies/PolicyLib/CMakeFiles/DptfPolicyLib.dir/home/leonard/software/dptf/DPTF/Sources/Policies/PolicyLib/ActiveTripPointsCachedProperty.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:689: Policies/PolicyLib/CMakeFiles/DptfPolicyLib.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

```

**Comment by leezu** [comment]
`[-Werror=catch-value=]` is fixed by https://github.com/intel/dptf/pull/24.

Please double check about the need for `-D_FORTIFY_SOURCE=2` in CMakeLists.txt.

**Comment by leezu** [comment]
Please also fix `-D_FORTIFY_SOURCE=2` and `-Werror=format-truncation=` in `ESIF/Products/ESIF_UF/Linux`

**Comment by bbian** [comment]
Commit f6983da26683086b3431f8b7c16528e7474ede25 should fix compilation issues for Ubuntu 18.04.
